00-programming-fundamentals/08-intro-to-automated-tests.md

```diff
@@ -60,24 +66,273 @@ true
+
+```ruby
+# die.rb
+```
+
+We can run the tests by typing:  `ruby die_test.rb` and get the following:
+
+```bash
+$ ruby die_test.rb
+Run options: --seed 35264
+
+# Running:
+
+
+
+Finished in 0.000751s, 0.0000 runs/s, 0.0000 assertions/s.
+
+0 runs, 0 assertions, 0 failures, 0 errors, 0 skips
+```
+
+We are now setup, but we haven't actually written a test yet.
+
+
+### A Tale of Two Styles
+
+In the TDD World there are two styles of testing.  In the first more traditional method people use *assertions* which are statements that check if a value is what it should be.  The other method is a subset of TDD, called Behavior-Driven Development (BDD) which accomplishes the same thing in a more English-friendly fashion that even non-developers can understand. 
+
+
+#### Writing Your First Test - Assertion Style
+
+To start we'll need to create a TestCase class.  We create a set of unit tests by subclassing MiniTest::Unit::TestCase and add each set of tests in instance methods.  
+
+
+```
+# die_test.rb
+
+require 'minitest/autorun'
+require 'minitest/reporters'
+require_relative 'die'
+
+class TestDie < MiniTest::Unit::TestCase
+  def test_creation_of_die
+    @die = Die.new
+    # The class of @die should be Die
+    assert @die.class == Die, "There must be a Die class."
+  end
+end
+```
+
+Notice the line `assert @die.class, Die`.  The `assert` method takes two required arguments and an optional message.  If the two required arguments are equal, the test passes, otherwise it fails.  
+
+When we run this with `ruby die_test.rb` we get:
+
+```bash
+ruby die_test.rb
+MiniTest::Unit::TestCase is now Minitest::Test. From die_test.rb:6:in `<main>'
+Run options: --seed 29986
+
+# Running:
+
+E
+
+Finished in 0.001219s, 820.3446 runs/s, 0.0000 assertions/s.
+
+  1) Error:
+TestDie#test_creation_of_die:
+NameError: uninitialized constant TestDie::Die
+Did you mean?  Dir
+    die_test.rb:8:in `test_creation_of_die'
+
+1 runs, 0 assertions, 0 failures, 1 errors, 0 skips
+```
+
+**This is a good thing.**  We have our first **red** test.  There's an error because we haven't created the Die class yet.
+
+Lets do so:
+
+```ruby
+# die.rb
+
+class Die
+
+end
+```
+
+Now `ruby die_test.rb` gets us:
+
+```bash
+MiniTest::Unit::TestCase is now Minitest::Test. From die_test.rb:6:in `<main>'
+Run options: --seed 61965
+
+# Running:
+
+.
+
+Finished in 0.001211s, 825.7638 runs/s, 825.7638 assertions/s.
+
+1 runs, 1 assertions, 0 failures, 0 errors, 0 skips
+```
+
+Now we have our first **green/passing** test.  
+
+There are a [number of assertions](https://gist.github.com/rastasheep/4248006#Minitest::Unit::TestCase) in Minitest beyond the assert method.  
+
+#### Practice Exercise
+
+Now we need to test that we can roll the die.  Write a test that checks the roll method of the Die class.  When a die is rolled it should return a number between 1 and 6 inclusive.  You can use the `assert_operator` method.  
```

Comment:
For this, they also need to write the method itself, right? The way it is worded right now it feels like it would get them to the "red" test but not necessarily writing the method in the Die class.